Sources/ArcGISToolkit/Components/Authentication/CertificatePickerViewModifier.swift

```diff
@@ -18,6 +18,16 @@ import ArcGIS
 /// An object that provides the business logic for the workflow of prompting the user for a
 /// certificate and a password.
 @MainActor final class CertificatePickerViewModel: ObservableObject {
+    /// The types of certificate error.
+    enum CertificateError: Error {
```

Comment:
Should this be public? Also, it seems a bit odd for it to be nested under the model.

Reply:
Only `Authenticator` and it's modifier are public, everything else is internal. The `CertificateError` is used within `CertificatePickerViewModel` and while displaying using `alertCertificateError`. Should it still be not nested?

Reply:
So users will never see an instance of `CertificateError`? If not, then it is fine where it is.

Reply:
Yes, they will see only error string in the alert.